MekHQ/src/mekhq/campaign/CampaignPreset.java

```diff
@@ -0,0 +1,496 @@
+import megamek.utils.MegaMekXmlUtil;
+import mekhq.MekHQ;
+import mekhq.MekHQOptions;
+import mekhq.MekHqConstants;
+import mekhq.MekHqXmlUtil;
+import mekhq.campaign.event.OptionsChangedEvent;
+import mekhq.campaign.personnel.SkillType;
+import mekhq.campaign.personnel.SpecialAbility;
+import mekhq.campaign.personnel.ranks.RankSystem;
+import mekhq.campaign.universe.Faction;
+import mekhq.campaign.universe.Factions;
+import mekhq.campaign.universe.Planet;
+import mekhq.campaign.universe.Systems;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+
+import javax.swing.*;
+import java.io.*;
+import java.nio.charset.StandardCharsets;
+import java.time.LocalDate;
+import java.util.*;
+
+/**
+ * This is an object which holds a set of objects that collectively define the initial options setup
+ * for a campaign.
+ *
+ * It includes both startup values, which are only used on initial startup (the date, starting
+ * planet, and rank system), and continuous options, which can be applied at any time (campaign
+ * options, skills, SPAs).
+ *
+ * It also includes a short title and description that allows one to create and save different
+ * presets. The goal is to allow users to create and load various different presets.
+ * @author Justin "Windchild" Bowen
+ */
+public class CampaignPreset implements Serializable {
+    //region Variable Declarations
+    private static final long serialVersionUID = 7753055687319002688L;
+
+    private final boolean userData;
+
+    private String title;
+    private String description;
+
+    // Startup
+    private LocalDate date;
+    private Faction faction;
+    private Planet planet;
+    private RankSystem rankSystem;
+    private int contractCount;
+
+    // Continuous
+    private GameOptions gameOptions;
+    private CampaignOptions campaignOptions;
+    private RandomSkillPreferences randomSkillPreferences;
+    private Hashtable<String, SkillType> skills;
+    private Hashtable<String, SpecialAbility> specialAbilities;
+    //endregion Variable Declarations
+
+    //region Constructors
+    public CampaignPreset() {
+        this(false);
+    }
+
+    public CampaignPreset(final boolean userData) {
+        this("Title", "", userData, null, null, null, null,
+                2, null, null, null,
+                new Hashtable<>(), new Hashtable<>());
+    }
+
+    public CampaignPreset(final Campaign campaign) {
+        this(campaign.getName(), "", true, campaign.getLocalDate(), campaign.getFaction(),
+                campaign.getCurrentSystem().getPrimaryPlanet(), campaign.getRankSystem(), 2,
+                campaign.getGameOptions(), campaign.getCampaignOptions(),
+                campaign.getRandomSkillPreferences(), SkillType.getSkillHash(),
+                SpecialAbility.getAllSpecialAbilities());
+    }
+
+    public CampaignPreset(final String title, final String description, final boolean userData,
```

Comment:
Nothing says "readable" like a constructor with 13 parameters, but it's either this or set all the properties individually which has its own set of problems.